packages/editor/src/lib/utils/richText.ts

```diff
@@ -39,34 +40,53 @@ export type RichTextFontVisitor = (
 	addFont: (font: TLFontFace) => void
 ) => RichTextFontVisitorState
 
+const fontsFromTextCache = new WeakCache<TLRichText, TLFontFace[]>()
+
+/**
+ * Whether a rich text object represents empty text.
+ *
+ * @param richText - The rich text content.
+ *
+ * @public
+ */
+export function isEmptyRichText(richText: TLRichText) {
+	if (richText.content.length === 1) {
+		if (!(richText.content[0] as any).content) return true
+	}
+	return false
+}
+
 /** @public */
 export function getFontsFromRichText(
 	editor: Editor,
 	richText: TLRichText,
 	initialState: RichTextFontVisitorState
 ) {
-	const { tipTapConfig, addFontsFromNode } = editor.getTextOptions()
-	assert(tipTapConfig, 'textOptions.tipTapConfig must be set to use rich text')
-	assert(addFontsFromNode, 'textOptions.addFontsFromNode must be set to use rich text')
+	if (isEmptyRichText(richText)) return EMPTY_ARRAY
+	return fontsFromTextCache.get(richText, () => {
```

Comment:
I don't think this is quite right - the fonts found will depend largely on `initialState`, which encodes stuff like the initial font (from the styles panel). Right now, i think if you change the font, you won't invalidate the cache.

Reply:
You can probably see this by removing the font preloading that we do in `onMount` in `Tldraw`